src/stories/Dialog.stories.tsx

```diff
@@ -79,12 +55,21 @@ const TemplateOption: StoryFn<DialogProps> = (args) => {
       <Dialog
         {...args}
         visibility={dialogVisibility}
-        rejectFunction={() => {
-          setDialogVisibility(!dialogVisibility);
-        }}
-        acceptFunction={() => {
-          setDialogVisibility(!dialogVisibility);
-        }}
+        setVisibility={setDialogVisibility}
+        rejectFunction={
+          !templateMessage
+            ? () => {
+                setDialogVisibility(!dialogVisibility);
+              }
+            : undefined
+        }
+        acceptFunction={
+          !templateMessage
+            ? () => {
+                setDialogVisibility(!dialogVisibility);
+              }
+            : undefined
+        }
```

Comment:
Since the purpose of this code is to serve as a reference, as a simple way to use the component, it might be better to create a conditional rendering of the entire Dialog component, and not repeat the condition for each property.
There is also the fact that setVisibility is not required when using rejectFunction and acceptFunction

Reply:
> Since the purpose of this code is to serve as a reference, as a simple way to use the component, it might be better to create a conditional rendering of the entire Dialog component, and not repeat the condition for each property. There is also the fact that setVisibility is not required when using rejectFunction and acceptFunction

conditional props values removed to use a conditional rendering